File: backend/gestion_commerciale/src/main/java/com/moonsystem/gestion_commerciale/services/TierService.java
package com.moonsystem.gestion_commerciale.services;

import java.util.List;

import com.moonsystem.gestion_commerciale.dto.TierDto;
import com.moonsystem.gestion_commerciale.utils.PageResponse;

public interface TierService {

    TierDto save(TierDto dto);

    TierDto update(TierDto dto);

    TierDto findById(int id);

    TierDto findById(Integer id);

    boolean delete(Integer id);

    PageResponse<TierDto> search(String keyword, int page, int size);

    List<String> getAllTierType();

    List<TierDto> findTierByQualite(String qualite);
    List<String> getAllVillesMaroc();

    List<TierDto> getAllClient();
    List<TierDto> getAllFournisseur();
    Integer numberOfTiers();

}

File: frontend/gestion-commerciale/src/app/services/tier/tier.service.ts
import { Injectable } from '@angular/core';
import { TierDto, TiersService } from 'src/app/api-client';

@Injectable({
  providedIn: 'root'
})
export class TierService {

  constructor(private tierService:TiersService) {

   }

   creerTier(tier:TierDto){
    return this.tierService.createTier(tier);
   }

   supprimerTier(id:number){
    return this.tierService.deleteTier(id);
   }
   modfierTier(tier:TierDto){
    return this.tierService.updateTier(tier);
   }
   listerTiers(keyword:string,page:number,size:number){
    return this.tierService.search(keyword, page, size);
   }
   listerQualitesTiers(){
    return this.tierService.getAllTierType();
   }
   listerVillesMarocaines(){
    return this.tierService.getAllTierVilles();
   }
   findById(id:number){
    return this.tierService.findById(id);
   }
   getNumber(){
    return this.tierService.numberOfTiers();
   }
   findAllClient(){
    return this.tierService.getAllClient();
   }

   findAllFournisseur(){
    return this.tierService.getAllFournisseur();
   }
   listerAllTiers(){
    return this.tierService.getAllTiers();
   }

}
